Cluster setting integration test needs better cleanup

As of writing there are two files used by integration tests to test cluster settings:
- [40_10_cluster_settings_disk_threshold.yml](https://github.com/vvanholl/elasticsearch-prometheus-exporter/blob/aad9723a91848a12e1638799247b8c0a984327f9/src/test/resources/rest-api-spec/test/resthandler/40_10_cluster_settings_disk_threshold.yml)
- [40_20_cluster_settings_disk_watermark_bytes.yml](https://github.com/vvanholl/elasticsearch-prometheus-exporter/blob/aad9723a91848a12e1638799247b8c0a984327f9/src/test/resources/rest-api-spec/test/resthandler/40_20_cluster_settings_disk_watermark_bytes.yml)

Tests are run in random order and the issue happens when the 40_20 file is run before 40_10. Then the test fails with:

```
  1> [2019-01-22T12:55:27,968][INFO ][o.e.r.PrometheusRestHandlerClientYamlTestSuiteIT] [test {yaml=elasticsearch-prometheus-exporter/40_20_cluster_settings_disk_watermark_bytes/Cluster settings statistics: watermark}]: before test
  1> [2019-01-22T12:55:28,206][INFO ][o.e.r.PrometheusRestHandlerClientYamlTestSuiteIT] [test {yaml=elasticsearch-prometheus-exporter/40_20_cluster_settings_disk_watermark_bytes/Cluster settings statistics: watermark}]: after test
 ...
  1> [2019-01-22T12:55:32,630][INFO ][o.e.r.PrometheusRestHandlerClientYamlTestSuiteIT] [test {yaml=elasticsearch-prometheus-exporter/40_10_cluster_settings_disk_threshold/Cluster settings statistics: disk threshold}]: before test
  1> [2019-01-22T12:55:32,639][INFO ][o.e.r.PrometheusRestHandlerClientYamlTestSuiteIT] Stash dump on test failure [{
  1>   "stash" : {
  1>     "body" : {
  1>       "persistent" : {
  1>         "cluster.routing.allocation.disk.watermark.high" : "99.9%",
  1>         "cluster.routing.allocation.disk.watermark.low" : "99.9%"
  1>       },
  1>       "transient" : { }
  1>     }
  1>   }
  1> }]
  1> [2019-01-22T12:55:32,648][INFO ][o.e.r.PrometheusRestHandlerClientYamlTestSuiteIT] [test {yaml=elasticsearch-prometheus-exporter/40_10_cluster_settings_disk_threshold/Cluster settings statistics: disk threshold}]: after test
  2> REPRODUCE WITH: ./gradlew :integTestRunner -Dtests.seed=A631F47304E16F58 -Dtests.class=org.elasticsearch.rest.PrometheusRestHandlerClientYamlTestSuiteIT -Dtests.method="test {yaml=elasticsearch-prometheus-exporter/40_10_cluster_settings_disk_threshold/Cluster settings statistics: disk threshold}" -Dtests.security.manager=true -Dtests.locale=ar-SA -Dtests.timezone=Europe/Malta
FAILURE 0.03s | PrometheusRestHandlerClientYamlTestSuiteIT.test {yaml=elasticsearch-prometheus-exporter/40_10_cluster_settings_disk_threshold/Cluster settings statistics: disk threshold} <<< FAILURES!
   > Throwable #1: java.lang.AssertionError: Failure at [elasticsearch-prometheus-exporter/40_10_cluster_settings_disk_threshold:12]: persistent didn't match the expected value:
   >                     persistent: 
   >   cluster.routing.allocation.disk.watermark.high: unexpected but found [99.9%]
   >   cluster.routing.allocation.disk.watermark.low: unexpected but found [99.9%]
   >    at __randomizedtesting.SeedInfo.seed([A631F47304E16F58:2E65CBA9AA1D02A0]:0)
   >    at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:349)
   >    at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.test(ESClientYamlSuiteTestCase.java:323)
   >    at java.lang.Thread.run(Thread.java:748)
   > Caused by: java.lang.AssertionError: persistent didn't match the expected value:
   >                     persistent: 
   >   cluster.routing.allocation.disk.watermark.high: unexpected but found [99.9%]
   >   cluster.routing.allocation.disk.watermark.low: unexpected but found [99.9%]
   >    at org.elasticsearch.test.rest.yaml.section.MatchAssertion.doAssert(MatchAssertion.java:92)
   >    at org.elasticsearch.test.rest.yaml.section.Assertion.execute(Assertion.java:76)
   >    at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:339)
   >    ... 37 more
```
To fix this we need to explicitly clean the persistent cluster settings in the end of 40_20 file. Interestingly, I am running into this issue only when I back-port integration tests to 5.x release. It might be that in ES 6.x the testing libraries are cleaning the cluster state differently (may be better?) between individual tests.

--- a/src/test/resources/rest-api-spec/test/resthandler/40_20_cluster_settings_disk_watermark_bytes.yml
+++ b/src/test/resources/rest-api-spec/test/resthandler/40_20_cluster_settings_disk_watermark_bytes.yml
@@ -166,3 +166,13 @@
         \# \s TYPE \s es_cluster_routing_allocation_disk_watermark_flood_stage_bytes \s gauge
         (\n \# \s (HELP|TYPE).* | \s*)
         /
+
+  # Clean up the cluster state. See https://github.com/vvanholl/elasticsearch-prometheus-exporter/issues/154
+  - do:
+      cluster.put_settings:
+        body:
+          persistent:
+            cluster.routing.allocation.disk.watermark.low: null
+            cluster.routing.allocation.disk.watermark.high: null
+            cluster.routing.allocation.disk.watermark.flood_stage: null
+        flat_settings: true

